IO/BeagleBone/CANBusBBB.cs

```diff
@@ -86,6 +86,30 @@ private unsafe struct CANFrame
 			public fixed byte Data[8];
 		}
 
+		[StructLayout(LayoutKind.Explicit)]
+		private unsafe struct ExtendedCANFrame
+		{
+			[FieldOffset(0), MarshalAs(UnmanagedType.U4)]
+			public uint CANID;
+
+			[FieldOffset(4), MarshalAs(UnmanagedType.U1)]
+			public byte DataLength;
+
+			[FieldOffset(5), MarshalAs(UnmanagedType.U1)]
+			public byte Flags;
+
+			[FieldOffset(6), MarshalAs(UnmanagedType.U1)]
+			public byte Reserved0;
+
+			[FieldOffset(7), MarshalAs(UnmanagedType.U1)]
+			public byte Reserved1;
+
+			//The original source forced this to 8-byte alignment with __attribute__((aligned(8)))
+			//If something breaks, maybe fix it?
+			[FieldOffset(8), MarshalAs(UnmanagedType.ByValArray, SizeConst = 64)]
+			public fixed byte Data[64];
+		}
+
 		[DllImport("libc", SetLastError = true)]
```

Comment:
In the past, we have separated our `DllImport`s into different files (see our RPi I/O devices). We could consider doing the same here.

Reply:
We could, but it's also not essential here. Up to you if you want to go through that effort.